### PR TITLE
Migrate devfile cmd validation to validate pkg

### LINKED ISSUE #3703: Move devfile command validation to validate pkg
This is a code refactor issue.

Currently the devfile commands are validated in command.go during `odo push`. However, other features of devfile requires the usage of devfile commands like lifecycle events and since they dont require the usage of `odo push`, they're validated outside in validate pkg.

We should make the command validation uniform and move it out from only `odo push` to the validate pkg like devfile events/component validation in validate pkg

/area devfile

### PR DESCRIPTION
**Pull request by maysunfaisal**
**What type of PR is this?**
/kind cleanup
/kind code-refactoring

**What does does this PR do / why we need it**:
- migrates devfile command validation to validate pkg along with other devfile validation
- devfile commands are now validated anytime we parse devfile and not only during odo push when using the command
- cleaned up some code
- updated unit tests

**Which issue(s) this PR fixes**:

Fixes #3703 

**PR acceptance criteria**:

- [x] Unit test 

- [x] Integration test 

- [ ] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
- Make invalid changes to the devfile command and any odo command that parses the devfile should err out

### CODE REVIEW

File: pkg/devfile/adapters/common/utils_test.go [hunk 1]
```diff
@@ -337,67 +337,67 @@ func TestGetBootstrapperImage(t *testing.T) {
...
+// 		name            string
+// 		component       common.DevfileComponent
+// 		wantIsSupported bool
+// 	}{
+// 		{
+// 			name:            "Case 1: Container component",
+// 			component:       testingutil.GetFakeContainerComponent("comp1"),
+// 			wantIsSupported: true,
+// 		},
+// 		{
```
Comment (mik-dass) [inline pkg/devfile/adapters/common/utils_test.go:352]: these comments need to be removed

File: pkg/devfile/parse.go [hunk 2]
```diff
@@ -65,10 +65,6 @@ func ParseAndValidate(path string) (d parser.DevfileObj, err error) {
 
 	// odo specific validation on devfile content
 	err = validate.ValidateDevfileData(d.Data)
-	if err != nil {
-		return d, err
-	}
 
-	// Successful
-	return d, nil
+	return d, err
```
Comment (mik-dass) [inline pkg/devfile/parse.go:69]: I think these changes can be applied in the above functions as well

File: pkg/devfile/validate/commands.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import (
+	"strings"
+
+	"github.com/openshift/odo/pkg/devfile/parser/data/common"
+)
+
+// validateCommands validates all the devfile commands
+func validateCommands(commands map[string]common.DevfileCommand, components []common.DevfileComponent) (err error) {
+
+	// commandsMap := adapterCommon.GetCommandsMap(commands)
```
Comment (mik-dass) [inline pkg/devfile/validate/commands.go:12]: do we need this?

File: pkg/devfile/validate/commands.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+	// type must be exec or composite
+	if command.Exec == nil && command.Composite == nil {
+		return &UnsupportedOdoCommandError{commandId: command.GetID()}
+	}
+
+	// If the command is a composite command, need to validate that it is valid
+	if command.Composite != nil {
+		parentCommands := make(map[string]string)
+		// commandsMap := adapterCommon.GetCommandsMap(commands)
```
Comment (mik-dass) [inline pkg/devfile/validate/commands.go:39]: I think this comment can be removed

**Comment by mik-dass** [comment]
Tests seem to be failing on Travis
https://travis-ci.com/github/openshift/odo/jobs/383364891#L2065

**Comment by mik-dass** [comment]
CI seems to be failing
```
e "runtime"
         ✗  the command "mkdir" does not map to a container component
        
    to contain substring
        <string>: references an invalid command
```

**Comment by maysunfaisal** [comment]
> CI seems to be failing
> 
> ```
> e "runtime"
>          ✗  the command "mkdir" does not map to a container component
>         
>     to contain substring
>         <string>: references an invalid command
> ```

@mik-dass yep, I noticed this yesterday

**Review by mik-dass — APPROVED** [review]
/lgtm

**Review by johnmcollier — APPROVED** [review]
Readding the lgtm label due to rebase

/lgtm

**Pull request merged**